src/track/serato/beatgrid.cpp

```diff
@@ -0,0 +1,401 @@
+#include "track/serato/beatgrid.h"
+
+#include <QtEndian>
+
+#include "util/logger.h"
+
+namespace {
+
+mixxx::Logger kLogger("SeratoBeatGrid");
+constexpr quint16 kVersion = 0x0100;
+constexpr int kMarkerSizeID3 = 8;
+const QByteArray kSeratoBeatGridBase64EncodedPrefix = QByteArray(
+        "application/octet-stream\x00\x00Serato BeatGrid\x00",
+        24 + 2 + 15 + 1);
+
+QByteArray base64encode(const QByteArray& data, bool chopPadding) {
+    QByteArray dataBase64;
+
+    // A newline char is inserted at every 72 bytes of base64-encoded content.
+    // Hence, we can split the data into blocks of 72 bytes * 3/4 = 54 bytes
+    // and base64-encode them one at a time:
+    int offset = 0;
+    while (offset < data.size()) {
+        if (offset > 0) {
+            dataBase64.append('\n');
+        }
+        QByteArray block = data.mid(offset, 54);
+        dataBase64.append(block.toBase64(
+                QByteArray::Base64Encoding | QByteArray::OmitTrailingEquals));
+        offset += block.size();
+
+        if (chopPadding) {
+            // In case that the last block would require padding, Serato seems to
+            // chop off the last byte of the base64-encoded data
+            if (block.size() % 3) {
+                dataBase64.chop(1);
```

Comment:
Your argument is based on logic and the rules of valid base64 encoding, which not what Serato's data format is about :wink:

Do you have a sample file from Serato where this is the case? Otherwise I'd say we leave it as is. I did some extensive testing when I originally I originally implemented this, so I'd rather leave this as is.

Reply:
No, I have no Serato installation.

Ok, than please add your test results as a comment. 
Interesting is: 
Did, you have a test file where Serato adds a zero byte where two bytes would be sufficient?  
Does Serato except a zero byte when two bytes are sufficient?